[build] Lombok annotation processing silently skipped under maven-compiler-plugin 3.15.0

## Problem

Lombok no longer runs during compilation — `@Data` / `@Builder` / `@Getter` etc. generate nothing, so every consumer of a Lombok-annotated class fails to compile with "cannot find symbol" for `getXxx()` / `setXxx()` / `builder()`.

```
[ERROR] /xchange-core/.../SomeClass.java:[NN,M] cannot find symbol
  symbol:   method getFoo()
  location: variable instance of type org.knowm.xchange.SomeData
```

### Reproduce

```
mvn clean compile
```

Fails on JDK 21+ (where javac requires `-proc` to be explicit) and emits a deprecation warning on JDK 17 ("Annotation processing is enabled because one or more processors were found on the class path").

### Cause

The root pom declares Lombok only as a `provided` dependency (`pom.xml:385-391`), with no `<annotationProcessorPaths>` on the `maven-compiler-plugin` config (`pom.xml:462-471`). `maven-compiler-plugin` 3.13.0+ deprecates classpath-based processor discovery, and JDK 21+ refuses it outright. Per the official Lombok Maven setup (https://projectlombok.org/setup/maven), both the `provided` dependency *and* an explicit `<annotationProcessorPaths>` entry are required.

Two pre-existing inconsistencies compound the problem:
- `xchange-coinsph/pom.xml` carries a defensive local `<annotationProcessorPaths>` override (and incorrectly redeclares Lombok with `scope=compile`).
- `xchange-coinbase/pom.xml` redeclares Lombok with no scope, leaking it onto the compile classpath.

## Fix

Add `<annotationProcessorPaths>` for `${version.lombok}` to the root `maven-compiler-plugin` config so all ~95 modules inherit it. Keep the existing `provided`-scoped Lombok dependency — it's still required for source-time symbol resolution and IDE support. This matches the official Lombok Maven recipe verbatim, stays on Java 17 (no `<release>` change), and works identically on JDK 17 / 21 / 23.

Drop the now-redundant overrides in `xchange-coinsph` and the stray Lombok redeclaration in `xchange-coinbase`.

--- a/pom.xml
+++ b/pom.xml
@@ -467,6 +467,13 @@
                     <release>${version.java}</release>
                     <showDeprecation>true</showDeprecation>
                     <showWarnings>true</showWarnings>
+                    <annotationProcessorPaths>
+                        <path>
+                            <groupId>org.projectlombok</groupId>
+                            <artifactId>lombok</artifactId>
+                            <version>${version.lombok}</version>
+                        </path>
+                    </annotationProcessorPaths>
                 </configuration>
             </plugin>
 

--- a/xchange-coinbase/pom.xml
+++ b/xchange-coinbase/pom.xml
@@ -48,11 +48,6 @@
             <artifactId>xchange-core</artifactId>
             <version>${project.version}</version>
         </dependency>
-        <dependency>
-            <groupId>org.projectlombok</groupId>
-            <artifactId>lombok</artifactId>
-            <version>${version.lombok}</version>
-        </dependency>
     </dependencies>
 
 </project>

--- a/xchange-coinsph/pom.xml
+++ b/xchange-coinsph/pom.xml
@@ -51,12 +51,6 @@
             <version>${version.mockito}</version>
             <scope>test</scope>
         </dependency>
-        <dependency>
-            <groupId>org.projectlombok</groupId>
-            <artifactId>lombok</artifactId>
-            <version>${version.lombok}</version>
-            <scope>compile</scope>
-        </dependency>
 
         <dependency>
             <groupId>org.wiremock</groupId>
@@ -66,19 +60,6 @@
     </dependencies>
     <build>
         <plugins>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-compiler-plugin</artifactId>
-                <configuration>
-                    <annotationProcessorPaths>
-                        <path>
-                            <groupId>org.projectlombok</groupId>
-                            <artifactId>lombok</artifactId>
-                            <version>${version.lombok}</version>
-                        </path>
-                    </annotationProcessorPaths>
-                </configuration>
-            </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-failsafe-plugin</artifactId>

